src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -164,28 +140,15 @@ public override void CopyTo(IHtmlContentBuilder destination)
                 return;
             }
 
-            for (var i = 0; i < Buffer.Count; i++)
+            if (_isSingleContentSet)
             {
-                var entry = Buffer[i];
-                if (entry == null)
-                {
-                    continue;
-                }
+                CopyToCore(SingleContent, destination);
+                return;
```

Comment:
Give this method the same shape as `MoveTo()` i.e. remove this `return` and nest the `for` in an `else`.